### ISSUE TITLE
Docs: zsh doesn't accept square brackets

### ISSUE THREAD

**Issue by mahiki** [tconbeer/harlequin#554]
**Before Proceeding, please acknowledge**:
- [x] I have consulted the [Troubleshooting guide](https://harlequin.sh/docs/troubleshooting/index).
- [x] I have searched Issues and Discussions in this repo.
- [x] Feature requests should be initiated as Discussions. This is a bug report.

**Describe the bug**
PIPX install - Its only the docs that need mention of install syntax for zsh:

```sh
pipx install harlequin[trino,s3]
# zsh: no matches found: harlequin[trino,s3]

pipx install 'harlequin[trino,s3]'
#  installed package harlequin 1.20.0, installed using Python 3.12.3
#  These apps are now globally available
#   - harlequin
# done! ✨ 🌟 ✨
```

I know its sort of a pipx thing but I spent a bit of time trying to understand my install problem. 

If you just close this issue a google search will show the way, sorry I don't have time to post a PR.

**Comment by tconbeer** [comment]
Yeah, I have this same note in the sqlfmt docs, should def include for harlequin. 

![image](https://github.com/tconbeer/harlequin/assets/705906/0e160fbb-df26-416b-9c03-bc5c384e6d02)

**Comment by mahiki** [comment]
Ah. Thanks for the amazing product!
